src/system_data.jl

```diff
@@ -377,31 +423,154 @@ end
-    component_dict = remove_components!(T, data.components)
-    for component in values(component_dict)
-        remove_forecasts_with_component(data, component)
+    components = remove_components!(T, data.components)
+    foreach(x -> remove_component_time_series!(data, x), components)
+end
+
+function remove_component_time_series!(data::SystemData, component)
+    for (uuid, label) in get_time_series_uuids(component)
+        remove_time_series!(data.time_series_storage, uuid, get_uuid(component), label)
+    end
+end
+
+"""
+    remove_forecast!(data::SystemData, component, initial_time::Dates.DateTime, label::String)
+
+Remove the time series data for a component.
+"""
+function remove_forecast!(
+                          ::Type{T},
+                          data::SystemData,
+                          component::InfrastructureSystemsType,
+                          initial_time::Dates.DateTime,
+                          label::String,
+                         ) where T <: Forecast
+    type_ = forecast_external_to_internal(T)
+    forecast = get_forecast(type_, component, initial_time, label)
+    uuid = get_time_series_uuid(forecast)
+    # TODO: can this be atomic?
+    remove_forecast_internal!(type_, component, initial_time, label)
+    remove_time_series!(data.time_series_storage, uuid, get_uuid(component), label)
+end
+
+function get_time_series(data::SystemData, forecast::Forecast)
+    return get_time_series(data.time_series_storage, get_time_series_uuid(forecast))
+end
+
+function clear_forecasts!(data::SystemData)
+    clear_forecasts!(data.components)
+    clear_time_series!(data.time_series_storage)
+end
+
+function iterate_forecasts(data::SystemData)
+    Channel() do channel
+        for component in iterate_components_with_forecasts(data.components)
+            for forecast in iterate_forecasts(component)
+                time_series = get_time_series(data.time_series_storage,
+                                              get_time_series_uuid(forecast))
+                put!(channel, make_public_forecast(forecast, time_series))
+            end
+        end
     end
 end
 
-function remove_forecasts_with_component(data::SystemData, component)
-    uuid = get_uuid(component)
-    to_delete = [f for f in iterate_forecasts(data) if get_uuid(get_component(f)) == uuid]
-    for forecast in to_delete
-        remove_forecast!(data, forecast)
+function get_forecast_initial_times(
+                                    ::Type{T},
+                                    data::SystemData,
+                                    component::InfrastructureSystemsType
+                                   ) where T <: Forecast
+    forecast_type = forecast_external_to_internal(T)
+    return get_forecast_initial_times(forecast_type, component)
+end
+
+function get_forecasts_interval(data::SystemData)
+    initial_times = get_forecast_initial_times(data)
+    if length(initial_times) <= 1
+        return UNINITIALIZED_PERIOD
```

Comment:
Why return UNINITIALIZED_PERIOD?

Reply:
UNINITIALIZED_PERIOD = 0 seconds.  The interval is defined as the difference between initial times.  In order to have an interval greater than 0 you would have to have at least two initial times.  I suppose it would be better to use 0 explicitly here.

Reply:
ok

Reply:
Another important note:  we've never had code that verifies intervals are consistent. I'm wondering if we should remove the concept from the System.  It could only exist when you're generating 'split' forecasts on the fly.

Reply:
I prefer to validate it when creating the forecast object so we support using forecasts outside of the system.